rollup/rollupsyncservice/l1client_test.go

```diff
@@ -0,0 +1,73 @@
+package rollupsyncservice
+
+import (
+	"context"
+	"math/big"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/scroll-tech/go-ethereum"
+	"github.com/scroll-tech/go-ethereum/common"
+	"github.com/scroll-tech/go-ethereum/core/types"
+	"github.com/scroll-tech/go-ethereum/rlp"
+)
+
+func TestL1Client(t *testing.T) {
+	ctx := context.Background()
+	mockClient := &mockEthClient{}
+
+	scrollChainABI, err := scrollChainMetaData.GetAbi()
+	if err != nil {
+		t.Fatal("failed to get scroll chain abi", "err", err)
+	}
+	scrollChainAddress := common.HexToAddress("0x0123456789abcdef")
+	l1Client, err := newL1Client(ctx, mockClient, 11155111, scrollChainAddress, scrollChainABI)
+	assert.Nil(t, err, "Failed to initialize L1Client")
+
+	blockNumber, err := l1Client.getLatestFinalizedBlockNumber(ctx)
+	assert.Nil(t, err, "Error getting latest confirmed block number")
+	assert.Equal(t, uint64(36), blockNumber, "Unexpected block number")
+
+	logs, err := l1Client.fetchRollupEventsInRange(ctx, 0, blockNumber)
+	assert.Empty(t, logs, "Expected no logs from fetchRollupEventsInRange")
+	assert.Nil(t, err, "Error fetching rollup events in range")
```

Comment:
prefer checking error before checking logs

Reply:
yeah, fixed in [6f7b8ab](https://github.com/scroll-tech/go-ethereum/pull/515/commits/6f7b8abd6bc1f09f8637368e474495a36761c15b)